test/e2e/utils/pods/rtepod/rtepod.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// rtepod includes utilities to introspect the RTE pod - and the RTE pod only
+package rtepod
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	RTEExecutable = "resource-topology-exporter"
+
+	notificationFileOption = "--notify-file"
+)
+
+func FindMetricsPort(rtePod *corev1.Pod) (int, error) {
+	for _, envVar := range rtePod.Spec.Containers[0].Env {
+		if envVar.Name == "METRICS_PORT" {
+			val, err := strconv.Atoi(envVar.Value)
+			if err != nil {
+				return 0, err
+			}
+			return val, nil
+		}
+	}
+	return 0, fmt.Errorf("cannot find METRICS_PORT environment variable")
+}
+
+func FindRTEContainerName(rtePod *corev1.Pod) (string, error) {
+	for idx := 0; idx < len(rtePod.Spec.Containers); idx++ {
+		cnt := &rtePod.Spec.Containers[idx] // shortcut
+		for _, arg := range cnt.Command {
+			if strings.Contains(arg, RTEExecutable) {
```

Comment:
we still somehow related on hardcoded values, but I do not have a better solution :(

Reply:
this is true. Howver I believe this PR is still a worthy improvement because:

1. we depend on less values, and more and more stable: we don't change the executable name often (ort at all) and same is true for the name of the options
2. if we ever do, now the tests should break with a clear indication why, so we can fix/revisit